source/m2mconnectionhandlerpimpl.cpp

```diff
@@ -622,6 +621,8 @@ bool M2MConnectionHandlerPimpl::is_handshake_ongoing()
 
 void M2MConnectionHandlerPimpl::receive_handler()
 {
+    unsigned char recv_buffer[BUFFER_LENGTH];
```

Comment:
I guess this buffer should be moved deeper into method, into the scope using it to potentially reduce peak stack usage.